### ISSUE TITLE
zaya_server.cpp: lm_head uses full VOCAB=262272 logit copy to host per token — destroys inference throughput

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#64]
## Location
`tests/zaya_server.cpp` — `forward()` lambda (line ~190) and `src/zaya_engine.cpp` — `zaya_forward()`

## Description

Two implementations of the forward pass handle lm_head very differently:

### zaya_server.cpp (the "207 KB binary"):
```cpp
// GPU lm_head: fused single-kernel launch (replaces 64 separate launches)
...
lm_head_fused_kernel<<<VOCAB,256,0,st>>>(d_all_logits,d_hs,d_embed_gpu,H,VOCAB);
argmax_kernel<<<1,256,0,st>>>(d_all_logits,VOCAB,d_best_idx,d_best_val);
hipStreamSynchronize(st);
// Copy only the winning token index (4 bytes) instead of all 262K floats
int best = 0;
hipMemcpy(&best,d_best_idx,4,hipMemcpyDeviceToHost);
```

This is efficient — only 4 bytes transferred back. ✅

### zaya_engine.cpp (the library version):
```cpp
void zaya_forward(ZayaState* s, int token_id, float* logits_out) {
    ...
    moe_tiled_gemv<<<(VOCAB+WMMA_M-1)/WMMA_M,WMMA_THREADS,0,s->st>>>(d_all,s->d_hs,s->d_embed,VOCAB,H);
    hipStreamSynchronize(s->st);
    std::vector<__half> lh(VOCAB);
    hipMemcpy(lh.data(),d_all,(size_t)VOCAB*2,hipMemcpyDeviceToHost);
    for(int v=0;v<VOCAB;v++)logits_out[v]=__half2float(lh[v]);
}
```

This copies **262,272 × 2 bytes = 524,544 bytes** from GPU to host on every single token! For a 0.6B model generating at 69 tok/s, that's ~36 MB/s of useless PCIe bandwidth consumed just to shuffle logits to host — when the only thing the caller does is find the argmax.

### zaya_forward_batch makes it worse:
```cpp
std::vector<__half> lh(B * VOCAB);
hipMemcpy(lh.data(), d_lm_batch, (size_t)B * VOCAB * 2, hipMemcpyDeviceToHost);
```

For B=8, that's **4 MB** per batch call.

## Impact

- Massive PCIe bandwidth waste: 524 KB per token vs the 4 bytes actually needed
- Dominates inference latency, especially for batch inference
- Defeats the purpose of having GPU-side argmax kernels

## Fix

`zaya_forward()` should use the same GPU argmax pattern that `zaya_forward_greedy()` and `zaya_server.cpp` use: compute logits on GPU, run argmax on GPU, copy back only 4 bytes. Only fall back to a full logit copy when the caller explicitly requests full logit distribution (e.g., for sampling/temperature).

```cpp
// Efficient path (default):
void zaya_forward(ZayaState* s, int token_id, int* argmax_out) {
    ... GPU kernels ...
    moe_tiled_gemv<<<...>>>(d_lm_buf, s->d_hs, s->d_embed, VOCAB, H);
    argmax_kernel<<<1, 256, 0, s->st>>>(d_lm_buf, VOCAB, d_argmax, d_argmax_val);
    hipStreamSynchronize(s->st);
    hipMemcpy(argmax_out, d_argmax, 4, hipMemcpyDeviceToHost);
}
```
